Documentation/stage1-implementors-guide.md

```diff
@@ -0,0 +1,81 @@
+
+Entrypoints
+-----------
+
+### `rkt run` => "coreos.com/rocket/stage1/run"
+
+After `rkt run` prepares the stage1 ACI and per-app stage2 ACI file-systems beneath "/var/lib/rkt/containers/$uuid" and writes the Container Runtime Manifest to "/var/lib/rkt/containers/$uuid/container", it chdirs to "/var/lib/rkt/containers/$uuid" and executes the program specified by the "coreos.com/rocket/stage1/run" entrypoint within the the stage1 ACI rootfs at "/var/lib/rkt/containers/$uuid/stage1/rootfs".
+
+It is the responsibility of this entrypoint to consume the Container Runtime Manifest and execute the constituent apps in the appropriate environments as specified by the Container Runtime Manifest.
+
+The environment variable "RKT_LOCK_FD" contains the file descriptor number of the open directory handle for "/var/lib/rkt/containers/$uuid".  It is necessary that stage1 leave this file descriptor open and in its locked state for the duration of the `rkt run`.
+
+In the bundled rocket stage1 which includes systemd-nspawn and systemd, the entrypoint is a static Go program found at "/init" within the stage1 ACI rootfs.  The majority of its execution entails composing a systemd-nspawn argument list and writing systemd unit files for the constituent apps before executing systemd-nspawn.  Systemd-nspawn then boots the stage1 systemd with the just-written unit files for launching the contained apps.  The "/init" program is essentially a Container Runtime Manifest to systemd-nspawn + systemd.service translator.
+
+An alternative stage1 could forego systemd-nspawn and systemd altogether, or retain these and introduce something like novm or qemu-kvm for greater isolation by first starting a VM.  All that is required is an executable at the place indicated by the "coreos.com/rocket/stage1/run" entrypoint which knows how to apply the Container Runtime Manifest and prepared ACI file-systems to good effect.
+
+
+#### Arguments
+* --debug
+* --private-net
+
+
+### `rkt enter` => "coreos.com/rocket/stage1/enter"
+
+After `rkt enter` determines which container and application is to be entered, it chdirs to "/var/lib/rkt/containers/$uuid" and executes the program specified by the "coreos.com/rocket/stage1/enter" entrypoint within the stage1 ACI rootfs at "/var/lib/rkt/containers/$uuid/stage1/rootfs".
+
+In the bundled rocket stage1 the entrypoint is a statically-linked C program found at "/enter" within the stage1 ACI rootfs.  This program enters the namespaces of the systemd-nspawn container's PID 1 before executing the "/diagexec" program which then chroots into the specific application's rootfs loading the application's environment variables as well.
+
+An alternative stage1 would need to do whatever is appropriate for entering the application's environment created by the stage1's "coreos.com/rocket/stage1/run" entrypoint.
+
+#### Arguments
+
+1. app ID to enter
+2. cmd
+3. cmd arguments
+
+
+
+Examples
+--------
```

Comment:
The headings are all =, subheadings -, deeper headings are 3 or more #s.

Isn't that how it's supposed to work?  Do you prefer they all be #'s?